packages/zoe/src/zoeService/startInstance.js

```diff
@@ -0,0 +1,225 @@
+            });
+
+          // return the userSeat before the offerHandler is called
+          return userSeat;
+        },
+        makeNoEscrowSeat: (
+          initialAllocation,
+          proposal,
+          exitObj,
+          seatHandle,
+        ) => {
+          const { userSeat, notifier, zoeSeatAdmin } = makeZoeSeatAdminKit(
+            initialAllocation,
+            exitZoeSeatAdmin,
+            hasExited,
+            proposal,
+            zoeInstanceStorageManager.withdrawPayments,
+            exitObj,
+          );
+          zoeSeatAdmins.add(zoeSeatAdmin);
+          seatHandleToZoeSeatAdmin.init(seatHandle, zoeSeatAdmin);
+          return { userSeat, notifier, zoeSeatAdmin };
+        },
+      });
+      return instanceAdmin;
+    };
+
+    const instanceAdmin = makeInstanceAdmin();
+    zoeInstanceStorageManager.initInstanceAdmin(instance, instanceAdmin);
+
+    E(adminNode)
+      .done()
+      .then(
+        completion => instanceAdmin.exitAllSeats(completion),
+        reason => instanceAdmin.failAllSeats(reason),
+      );
+
+    /** @type {ZoeInstanceAdmin} */
+    const zoeInstanceAdminForZcf = Far('zoeInstanceAdminForZcf', {
+      makeInvitation: zoeInstanceStorageManager.makeInvitation,
+      // checks of keyword done on zcf side
+      saveIssuer: zoeInstanceStorageManager.saveIssuer,
+      // A Seat requested by the contract without any payments to escrow
+      makeNoEscrowSeat: instanceAdmin.makeNoEscrowSeat,
+      exitAllSeats: completion => instanceAdmin.exitAllSeats(completion),
+      failAllSeats: reason => instanceAdmin.failAllSeats(reason),
+      makeZoeMint: zoeInstanceStorageManager.makeZoeMint,
+      replaceAllocations: seatHandleAllocations => {
+        seatHandleAllocations.forEach(({ seatHandle, allocation }) => {
+          const zoeSeatAdmin = seatHandleToZoeSeatAdmin.get(seatHandle);
+          zoeSeatAdmin.replaceAllocation(allocation);
+        });
+      },
+      stopAcceptingOffers: () => instanceAdmin.stopAcceptingOffers(),
+    });
+
+    // At this point, the contract will start executing. All must be
+    // ready
+
+    const {
+      creatorFacet = Far('emptyCreatorFacet', {}),
+      publicFacet = Far('emptyPublicFacet', {}),
+      creatorInvitation: creatorInvitationP,
+      handleOfferObj,
+    } = await E(zcfRoot).executeContract(
+      bundle,
+      zoeServicePromise,
+      zoeInstanceStorageManager.invitationIssuer,
+      zoeInstanceAdminForZcf,
+      zoeInstanceStorageManager.exportInstanceRecord(),
+      zoeInstanceStorageManager.exportIssuerStorage(),
+    );
+
+    handleOfferObjPromiseKit.resolve(handleOfferObj);
+    publicFacetPromiseKit.resolve(publicFacet);
+
+    // creatorInvitation can be undefined, but if it is defined,
+    // let's make sure it is an invitation.
+    return Promise.allSettled([
+      creatorInvitationP,
+      zoeInstanceStorageManager.invitationIssuer.isLive(creatorInvitationP),
```

Comment:
The invitationIssuer comes from the zoeInstanceStorageManager now.

Reply:
understood. Resolving.

I am surprised that something named `fooManager` is not all methods.